Src/Microsoft.Dynamic/Ast/ConstantExpression.cs

```diff
@@ -55,26 +56,39 @@ public static partial class Utils {
                 return NullLiteral;
             }
 
-            BigInteger bi = value as BigInteger;
-            if ((object)bi != null) {
-                return BigIntegerConstant(bi);
-            } else if (value is BigInt) {
-                return BigIntConstant((BigInt)value);
-            } else if (value is Complex) {
+            if (value is BigInteger) {
```

Comment:
What about using a switch statement instead? Removing the redundant elses ends up taking something quite compact and makes it take a lot more space.